tap_zendesk/streams.py

```diff
@@ -21,52 +25,309 @@
+                    th.Property("address", th.StringType),
+                    th.Property("subject", th.StringType),
+                    th.Property("brand_id", th.StringType),
+                    th.Property("formatted_phone", th.StringType),
+                    th.Property("phone", th.StringType),
+                    th.Property("profile_url", th.StringType),
+                    th.Property("twitter_id", th.StringType),
+                    th.Property("username", th.StringType),
+                    th.Property("channel", th.StringType)
+                )),
+                th.Property("to", th.ObjectType(
+                    th.Property("address", th.StringType),
+                    th.Property("name", th.StringType),
+                    th.Property("brand_id", th.StringType),
+                    th.Property("formatted_phone", th.StringType),
+                    th.Property("phone", th.StringType),
+                    th.Property("profile_url", th.StringType),
+                    th.Property("twitter_id", th.StringType),
+                    th.Property("username", th.StringType)
+                )),
+                th.Property("rel", th.StringType)
+            )),
+            th.Property("channel", th.StringType)
+        )),
+        th.Property("ticket_form_id", th.IntegerType),
+        th.Property("satisfaction_rating", th.ObjectType(
+            th.Property("id", th.IntegerType),
+            th.Property("assignee_id", th.IntegerType),
+            th.Property("group_id", th.IntegerType),
+            th.Property("reason_id", th.IntegerType),
+            th.Property("requester_id", th.IntegerType),
+            th.Property("ticket_id", th.IntegerType),
+            th.Property("updated_at", th.DateTimeType),
+            th.Property("created_at", th.DateTimeType),
+            th.Property("url", th.StringType),
+            th.Property("score", th.StringType),
+            th.Property("reason", th.StringType),
+            th.Property("comment", th.StringType)
+        )),
+        th.Property("sharing_agreement_ids", th.ArrayType(th.IntegerType)),
+        th.Property("email_cc_ids", th.ArrayType(th.IntegerType)),
+        th.Property("forum_topic_id", th.IntegerType)
     ).to_dict()
 
+    def get_child_context(self, record: dict, context: Optional[dict]) -> dict:
+        """Return a context dictionary for child streams."""
+        self.logger.debug(f"Creating child context for ticket_id: {record['id']}")
+        return {
+            "ticket_id": record["id"],
+        }
 
-class GroupsStream(ZendeskStream):
-    """Define custom stream."""
 
-    name = "groups"
-    path = "/groups"
-    primary_keys: t.ClassVar[list[str]] = ["id"]
-    replication_key = "modified"
+class TicketCommentsStream(NonIncrementalZendeskStream):
+    name = "ticket_comments"
+    parent_stream_type = TicketsStream
+    path = "/api/v2/tickets/{ticket_id}/comments.json"
+    primary_keys = ["id"]
+    replication_key = "created_at"
```

Comment:
We need to check if the `created_at` will be correctly defined because we can't guarantee the order since this stream is collected from each ticket. 
In theory it should use only the `replication_key` from the parent stream.

Reply:
In my testing, with or without specifying `created_at` here, we are getting the same results, so I have removed it.